docs/products/opensearch/howto/opensearch-search-and-python.rst

```diff
@@ -0,0 +1,439 @@
+                      }
+                    }
+                  }
+
+If you're looking for ``pannacotta marmalade`` phrase. To find more results rather than exact phrases, you should allow a certain degree like setting the ``slop=2``, so it can find matches skipping two words between the searched ones:
+
+.. code-block:: shell
+
+  python search.py slop "title" "pannacotta marmalade" 2
+
+
+With this flexibility, you can find titles with the desired words even if there are other words in between all thanks to the ``slop`` parameter.
+
+.. code-block:: python
+
+    ['Lemon Pannacotta with Lemon Marmalade ']
+
+
+Try out the ``slop`` query from our `demo <https://github.com/aiven/demo-opensearch-python>`__ to allow some flexibility in your search query:
+
+.. seealso::
+
+  Read more about ``slop`` parameter on the `OpenSearch project specifications <https://opensearch.org/docs/latest/opensearch/query-dsl/full-text#options>`_.
+
+
+Term query
+----------
+If you're looking to find in a ``field`` an exact ``value``, the `term query <https://opensearch.org/docs/latest/opensearch/query-dsl/term/#term>`_ is the right choice. This query can be constructed as:
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "term": {
+                        field: value
+                      }
+                    }
+                  }
+
+
+Curious about recipes low in sodium? 
+
+You can find out more recipes with ``term`` queries by running the `demo <https://github.com/aiven/demo-opensearch-python>`__ application:
+
+::
+
+  python search.py term sodium 0
+
+
+Range query
+-----------
+
+This query helps to find documents that the searched field's value is within a certain range. This can be handy if you're dealing with numerical values and are interested in ranges instead of specific values. The queries can be constructed as:
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "range": {
+                        field: {
+                          "gte": gte,
+                          "lte": lte
+                        }
+                      }
+                    }
+                  }
+
+You can construct range queries with combinations of inclusive and exclusive parameters as can be seen in the table:
+
+.. list-table::
+  :header-rows: 1
+  :stub-columns: 1
+  :align: left
+
+  * - Parameter
+    - Behavior
+  * - ``gte``
+    - Greater than or equal to
+  * - ``gt``
+    - Greater than
+  * - ``lt``
+    - Less than
+  * - ``lte``
+    - Less than or equal to
+
+Check out using our `demo <https://github.com/aiven/demo-opensearch-python>`__ which recipes you can find within a certain range of sodium, for example:
+
+::
+
+    python search.py range sodium 0 10
+
+Fuzzy queries
+-------------
+You can look for fuzzy combinations where variations of the words are allowed, also called expansions, returning the exact matches for those expansions. 
```

Comment:
This is pretty long. Maybe split it into couple of sentences. And I'd give a short intro on what are fuzzy queries and why they are useful.

Reply:
done!